### DISCUSSION TITLE
Using the colorpicker outside of the window

### DISCUSSION THREAD

**Discussion by ChaonicTheDeathKitten** [godotengine/godot-proposals#9187] [Editor]
Hello!
One problem I seem to run into everywhere is that I have to copy individual color codes one by one, which can sometimes be just five, sometimes up to forty or more. One feature some programs have that make my life immensely easier is when I can use the colorpicker outside the window. Photoshop does it, [Aseprite added it last year](https://twitter.com/aseprite/status/1397596285587410956) and [Blender got an update receiving that functionality 5 months ago aswell](https://blender.community/c/rightclickselect/Kkbbbc/?sorting=hot).

I personally think it would be a good fit for the Godot editor. Aswell as art applications made with the Godot engine.
Admittedly, the whole reason I'm this invested is not because I've designed an art tool I'm dying to make in Godot and it hinges entirely on that feature. And no matter how I look at it, I don't think I'm remotely up to the task making this kind of contribution to the engine.

But anyways, what are your thoughts on this?

**Comment by Calinou** [comment]
This is already possible as of Godot 4.2.1 (possibly in earlier versions starting from 4.0). Here's an example on Linux:

https://github.com/godotengine/godot-proposals/assets/180032/96a3bb08-3e74-4a42-891f-75c031b60366

This feature works in projects using the ColorPicker(Button) node as well.

We could improve on this by making the OS ignore the click that picks the color (so you don't trigger buttons in other programs when clicking one while picking a color).

**Reply by ChaonicTheDeathKitten** [reply]
You're right! It IS possible! I feel like it wasn't even two months ago that I have played around with it and it wasn't a thing yet.
And since googling and searching for the feature on here didn't result in anything, I assumed wrong.
This is great!
I've just had a good look at it and its documentation. Is it at all possible to create my own button or call for the color sampler to be active in code? This is already pretty much all I need, I'm just wondering if I'd be able to additionally create a hotkey.

And yes, you're right, this addition could be really helpful! I've on more than a couple occasions taken still frames from videos as a source for color palettes. And moving on in the video would possibly mess up my source! I would love this addition!

This makes me so happy! Thank you!!!

**Reply by bruvzg** [reply]
> Is it at all possible to create my own button or call for the color sampler to be active in code?

You can read screen color at the mouse position using the following code:
```gdscript
var color = DisplayServer.screen_get_pixel(DisplayServer.mouse_get_position())
```

**Reply by ChaonicTheDeathKitten** [reply]
You people are amazing!
Thanks a bunch! <3
